src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs

```diff
@@ -67,7 +69,7 @@ private LambdaCompiler(AnalyzedTree tree, LambdaExpression lambda)
         {
             Type[] parameterTypes = GetParameterTypes(lambda, typeof(Closure));
 
-            var method = new DynamicMethod(lambda.Name ?? "lambda_method", lambda.ReturnType, parameterTypes, true);
+            var method = new DynamicMethod(lambda.Name ?? ("lambda_method" + s_lambdaMethodIndex.ToString()), lambda.ReturnType, parameterTypes, true);
```

Comment:
Maybe this was supposed to be:
```suggestion
            var method = new DynamicMethod(lambda.Name ?? ("lambda_method" + Interlocked.Increment(ref s_lambdaMethodIndex).ToString()), lambda.ReturnType, parameterTypes, true);
```
?